editor/src/messages/portfolio/document/utility_types/network_interface.rs

```diff
@@ -1124,55 +1136,39 @@ impl NodeNetworkInterface {
 		Some(&node.implementation)
 	}
 
-	pub fn input_name<'a>(&'a self, node_id: NodeId, index: usize, network_path: &[NodeId]) -> Option<&'a str> {
-		let Some(input_row) = self.input_properties_row(&node_id, index, network_path) else {
-			log::error!("Could not get input_name for node {node_id} index {index}");
-			return None;
-		};
-		let name = input_row.input_name.as_str();
-		if !name.is_empty() {
-			Some(name)
-		} else {
-			let node_definition = resolve_document_node_type(self.reference(&node_id, network_path)?.as_ref()?)?;
-			let rows = &node_definition.node_template.persistent_node_metadata.input_properties;
-
-			rows.get(index).map(|row| row.input_name.as_str())
-		}
-	}
-
-	pub fn input_description<'a>(&'a self, node_id: NodeId, index: usize, network_path: &[NodeId]) -> Option<&'a str> {
-		let Some(input_row) = self.input_properties_row(&node_id, index, network_path) else {
-			log::error!("Could not get input_row in input_description");
-			return None;
-		};
-		let description = input_row.input_description.as_str();
-		if !description.is_empty() && description != "TODO" {
-			Some(description)
-		} else {
-			let node_definition = resolve_document_node_type(self.reference(&node_id, network_path)?.as_ref()?)?;
-			let rows = &node_definition.node_template.persistent_node_metadata.input_properties;
-
-			rows.get(index).map(|row| row.input_description.as_str())
-		}
+	pub fn input_data(&self, node_id: &NodeId, index: usize, key: &str, network_path: &[NodeId]) -> Option<&Value> {
+		let metadata = self
+			.node_metadata(node_id, network_path)
+			.and_then(|node_metadata| node_metadata.persistent_metadata.input_metadata.get(index))?;
+		metadata.persistent_metadata.input_data.get(key)
 	}
-
-	pub fn input_properties_row(&self, node_id: &NodeId, index: usize, network_path: &[NodeId]) -> Option<&PropertiesRow> {
-		self.node_metadata(node_id, network_path)
-			.and_then(|node_metadata| node_metadata.persistent_metadata.input_properties.get(index))
+	pub fn persistent_input_metadata(&self, node_id: &NodeId, index: usize, network_path: &[NodeId]) -> Option<&InputPersistentMetadata> {
+		let metadata = self
+			.node_metadata(node_id, network_path)
+			.and_then(|node_metadata| node_metadata.persistent_metadata.input_metadata.get(index))?;
+		Some(&metadata.persistent_metadata)
 	}
 
-	pub fn insert_input_properties_row(&mut self, node_id: &NodeId, index: usize, network_path: &[NodeId], row: PropertiesRow) {
```

Comment:
this should never be called, since when upgrading a node the inputs and metadata are added at the same time to ensure they stay in sync